staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -1054,6 +1059,10 @@ func mapsHaveConflicts(typedLeft, typedRight map[string]interface{}, structType
 					return true, err
 				}
 
+				if fieldPatchStrategy == replaceDirective {
```

Comment:
is there a reason not to push this into mergingMapFieldsHaveConflicts? if the strategy is replace, it should return false, right?